src/back/msl/mod.rs

```diff
@@ -204,8 +204,20 @@ impl ResolvedBinding {
     }
 }
 
-pub fn write_string(module: &crate::Module, options: &Options) -> Result<String, Error> {
+/// Information about a translated that is required
+/// for the use of the result.
+pub struct TranslationInfo {
+    /// Mapping of the entry point names. Each item in the array
+    /// corresponds to an entry point in `module.entry_points.iter()`.
+    pub entry_point_names: Vec<String>,
```

Comment:
nit: `HashMap<(String, ExecutionModel), String>)` might be more convenient instead of iterating to find the index of our entry point

Reply:
I'm not sure about this. With the way this PR sets it up, gfx-backend-metal doesn't have to do any work at all, i.e. this vector is most convenient. Switching it to a hashmap would be more user-oriented (versus gfx backend oriented), and would require more code on the gfx side, while the benefit isn't clear.